sdk/python/feast/loaders/producer_interface.py

```diff
@@ -0,0 +1,126 @@
+from typing import Optional, Union
+
+
+class ProducerInterface:
+    """
```

Comment:
Should this maybe just be an Producer ABC?

Reply:
I tried naming this as `Producer`. I recall running into import errors because Python was not able to resolve the ambiguity of `from confluent_kafka import Producer` if i recall correctly.

Reply:
AbstractProducer?

Reply:
Renamed!